widgetastic_manageiq/__init__.py

```diff
@@ -2357,42 +2353,45 @@ def get_entities(self, by_name=None, surf_pages=False):
         entities = self.get_all(surf_pages)
         remaining_entities = []
         for entity in entities:
-            if by_name and by_name in entity.name:
+            if by_name in entity.name:
                 remaining_entities.append(entity)
             # todo: by_type and by_regexp will be implemented later if needed
+        if not remaining_entities:
+            raise ItemNotFound("Entities matching to {name} "
+                               "aren't found on this page".format(name=by_name))
         return remaining_entities
 
-    def get_entity(self, by_name=None, surf_pages=False):
-        """ obtains one entity matched to by_name
-        raises exception if no entities or several entities were found
-        Args:
-            by_name (str): only entity which match to by_name will be returned
-            surf_pages (bool): current page entity if False, all entities otherwise
-
-        Returns: matched entities (QuadIcon/etc.)
-        """
-        entities = self.get_entities(by_name=by_name, surf_pages=surf_pages)
-        if len(entities) == 0:
-            raise ItemNotFound("Entity {name} isn't found on this page".format(name=by_name))
-        elif len(entities) > 1:
-            raise ManyEntitiesFound("Several entities with {name} were found".format(name=by_name))
-        return entities[0]
-
-    def get_first_entity(self, by_name=None):
+    def get_entity(self, by_name, surf_pages=False):
         """ obtains one entity matched to by_name and stops on that page
-        raises exception if no entity or several entities were found
         Args:
             by_name (str): only entity which match to by_name will be returned
+            surf_pages (bool): current page entity if False, all entities otherwise
 
         Returns: matched entity (QuadIcon/etc.)
         """
         for _ in self.paginator.pages():
             found_entities = [self.parent.entity_class(parent=self, name=name)
                               for name in self.entity_names if by_name == name]
-            if found_entities:
+            if len(found_entities) == 1:
                 return found_entities[0]
+            elif len(found_entities) > 1:
+                raise ManyEntitiesFound("Several entities with "
+                                        "{name} were found".format(name=by_name))
```

Comment:
I see the flow now, the for's else catches. on no return, and found_entities isn't used. looks safe.